packages/scaffold/src/views/w3m-transactions-view/index.ts

```diff
@@ -0,0 +1,219 @@
+            ${this.templateTransactions(transactions, isLastGroup)}
+          </wui-flex>
+        </wui-flex>
+      `
+    })
+  }
+
+  private templateTransactions(transactions: Transaction[], isLastGroup: boolean) {
+    return transactions.map((transaction, index) => {
+      const { date, descriptions, direction, images, status, type } =
+        this.getTransactionListItemProps(transaction)
+      const isLastTransaction = isLastGroup && index === transactions.length - 1
+
+      return html`
+        <wui-transaction-list-item
+          date=${date}
+          direction=${direction}
+          id=${isLastTransaction && this.next ? PAGINATOR_ID : ''}
+          status=${status}
+          type=${type}
+          .images=${images}
+          .descriptions=${descriptions}
+        ></wui-transaction-list-item>
+      `
+    })
+  }
+
+  private templateEmpty() {
+    return html`
+      <wui-flex
+        flexGrow="1"
+        flexDirection="column"
+        justifyContent="center"
+        alignItems="center"
+        .padding=${['3xl', 'xl', '3xl', 'xl'] as const}
+        gap="xl"
+      >
+        <wui-icon-box
+          backgroundColor="glass-005"
+          background="gray"
+          iconColor="fg-200"
+          icon="wallet"
+          size="lg"
+          ?border=${true}
+          borderColor="wui-color-bg-125"
+        ></wui-icon-box>
+        <wui-flex flexDirection="column" alignItems="center" gap="xs">
+          <wui-text align="center" variant="paragraph-500" color="fg-100"
+            >No Transactions yet</wui-text
+          >
+          <wui-text align="center" variant="small-500" color="fg-200"
+            >Start trading on dApps <br />
+            to grow your wallet!</wui-text
+          >
+        </wui-flex>
+      </wui-flex>
+    `
+  }
+
+  private templateLoading() {
+    return Array(LOADING_ITEM_COUNT)
+      .fill(null)
+      .map(() => html` <wui-transaction-list-item-loader></wui-transaction-list-item-loader> `)
+  }
+
+  private createPaginationObserver() {
+    this.paginationObserver = new IntersectionObserver(([element]) => {
+      if (element?.isIntersecting && !this.loading) {
+        this.fetchTransactions()
+        EventsController.sendEvent({ type: 'track', event: 'LOAD_MORE_TRANSACTIONS' })
+      }
+    }, {})
+    this.setPaginationObserver()
+  }
+
+  private setPaginationObserver() {
+    this.paginationObserver?.disconnect()
+
+    const lastItem = this.shadowRoot?.querySelector(`#${PAGINATOR_ID}`)
+    if (lastItem) {
+      this.paginationObserver?.observe(lastItem)
+    }
+  }
+
+  private async fetchTransactions() {
+    await TransactionsController.fetchTransactions()
+  }
+
+  private getTransactionListItemProps(transaction: Transaction) {
+    const date = DateUtil.getRelativeDateFromNow(transaction?.metadata?.minedAt)
+    const descriptions = TransactionUtil.getTransactionDescriptions(transaction)
+
+    const transfer = transaction?.transfers?.[0]
```

Comment:
```ts
const [transfer, secondTransfer] = transaction?.transfers ?? [null, null]
```

Reply:
We don't need second transfer here now, so skipping this